website/src/blogs/openebs-node-device-management-ndm-e2-80-8a--e2-80-8atroubleshooting-tips.md

```diff
@@ -0,0 +1,120 @@
+
+**Symptom:** When a node in the cluster rebooted. A blockdevice resource on that node was marked as inactive and a new resource was created. The new blockdevice also has the same details as the old one.
+
+**Troubleshooting:**
+
+1. Check `lsblk` output of the node
+2. Get the yaml of both blockdevices and compare them.
+3. Check to see `spec.Path` is different in both outputs.
+4. If yes, then the new blockdevice resource was created because the path changed
+5. Check if `kubernetes.io/hostname` is different, if yes , then the blockdevice was created because the hostname of the node changed.
+
+**Resolution:** If using cStor, the newly generated BD can be added in both SPC and CSP instead of the old BD resource. Thus the storage engine will claim the new BD resource and start using it.
+
+**Root Cause:** Whenever the NDM deamonset pods shutdown, all the devices on that node will be marked into an UNknown state. When the pod comes backup, all the devices on that node are marked as inactive and then individual devices are processed for their statuses.
+
+NDM uses an md5 sum of WWN+Model+Serial of the disk to create its unique name. If none of these fields are available then NDM uses device path and hostname to create the blockdevice. There are chances that the device path/hostname has changed after reboot. If the path/hostname changes a new blockdevice resource will be created, and the old one will still be in the inactive state.
+
+---
+
+#### Scenario #3
+
+**BlockDevices are created for already used disks in which OS is installed**
+
+**Symptom:** NDM created blockdevice resources for disks which are already used for OS partitions. By default NDM excludes the blockdevices that are mounted at `/, /boot, /etc/hosts`. If these mount points are on an LVM or SoftRaid, NDM will not be able to identify that.
+
+**Resolution:** Support for LVM and software RAID is in the design phase. Once it is supported the issue will be resolved.
+
+---
+
+#### Scenario #4
+
+**Only one Blockdevice is created, when devices are connected in multipath configuration**
+
+**Symptom:** There is a disk attached in multipath configuration to a node. i.e both sdb & sdc are the same devices. But blockdevice resource is created only for sdc.
+
+**Resolution:** Support for detecting disks in multipath configuration and attaching the same disk to multiple nodes will be available in the future versions of NDM
+
+**Root Cause:** NDM generates the UID for disk identification using the disk details like WWN, Serial etc that are fetched from the disk. In case of a disk attached in multipath configuration, the details from both sdb and sdc will be the same. Therefore, NDM will first create a blockdevice for sdb, and then moves on to create for `sdc`. But at this stage it will find that a blockdevice with that UID already exists and will update the blockdevice information with the new path `sdc`. This results in a blockdevice existing only for sdc.
```

Comment:
```suggestion
**Root Cause:** NDM generates the UID for disk identification using the disk details like WWN, Serial, etc that are fetched from the disk. In case of a disk attached in multipath configuration, the details from both sdb and sdc will be the same. Therefore, NDM will first create a blockdevice for sdb, and then moves on to create for `sdc`. But at this stage, it will find that a blockdevice with that UID already exists and will update the blockdevice information with the new path `sdc`. This results in a blockdevice existing only for sdc.
```